qemu/tests/virtio_fs_share_data.py

```diff
@@ -104,24 +112,36 @@ def get_viofs_exe(session):
     cmd_set_tmpfs = params.get('cmd_set_tmpfs')
     size_mem1 = params.get('size_mem1')
 
-    # xfstest-nfs config
+    # nfs config
     setup_local_nfs = params.get('setup_local_nfs')
 
+    # st_dev check config
+    cmd_get_stdev = params.get("cmd_get_stdev")
+    nfs_mount_dst_name = params.get("nfs_mount_dst_name")
+
     if cmd_xfstest:
         # /dev/shm is the default memory-backend-file, the default value is the
         # half of the host memory. Increase it to guest memory size to avoid crash
         ori_tmpfs_size = process.run(cmd_get_tmpfs, shell=True).stdout_text.replace("\n", "")
         logging.debug("original tmpfs size is %s", ori_tmpfs_size)
         params["post_command"] = cmd_set_tmpfs % ori_tmpfs_size
         params["pre_command"] = cmd_set_tmpfs % size_mem1
-        if setup_local_nfs:
-            for fs in params.objects("filesystems"):
-                nfs_params = params.object_params(fs)
-                params["export_dir"] = nfs_params.get("export_dir")
-                params["nfs_mount_src"] = nfs_params.get("nfs_mount_src")
-                params["nfs_mount_dir"] = nfs_params.get("fs_source_dir")
-                nfs_local = nfs.Nfs(params)
-                nfs_local.setup()
+        if not setup_local_nfs:
+            params["start_vm"] = "yes"
+            env_process.preprocess(test, params, env)
+
+    if setup_local_nfs:
+        for fs in params.objects("filesystems"):
+            nfs_params = params.object_params(fs)
+
+            params["export_dir"] = nfs_params.get("export_dir")
+            params["nfs_mount_src"] = nfs_params.get("nfs_mount_src")
+            params["nfs_mount_dir"] = nfs_params.get("fs_source_dir")
+            if cmd_get_stdev:
+                fs_source_dir = nfs_params.get("fs_source_dir")
+                params["nfs_mount_dir"] = os.path.join(fs_source_dir, nfs_mount_dst_name)
+            nfs_local = nfs.Nfs(params)
+            nfs_local.setup()
```

Comment:
The original codes is `env_process.preprocess(test, params, env) ` if `cmd_xfstest` is True,so we better to keep it.
I suggest that:
```

if cmd_xfstest:
    ...
if setup_local_nfs:
    ...
if cmd_xfstest or setup_local_nfs:
    params["start_vm"] = "yes"
    env_process.preprocess(test, params, env)    

```
This looks like more readable.